precompile/contracts/warp/predicate_test.go

```diff
@@ -212,7 +233,8 @@ func createValidPredicateTest(snowCtx *snow.Context, numKeys uint64, predicate p
 			},
 		},
 		Predicate:   predicate,
-		Gas:         gasConfig.PerSignatureVerification + uint64(len(predicate))*gasConfig.PerWarpMessageChunk + numKeys*gasConfig.PerWarpSigner,
+		Rules:       graniteRules,
+		Gas:         graniteGasConfig.PerSignatureVerification + uint64(len(predicate))*graniteGasConfig.PerWarpMessageChunk + numKeys*graniteGasConfig.PerWarpSigner,
```

Comment:
Please abstract a helper function to calculate this. Something like:

```go
func predicateGasCost(c *GasConfig, chunks, signers int) uint64

// or

func (c *GasConfig) predicateGasCost(chunks, signers int) uint64
```

I recommend accepting `int` instead of `uint64` so you can pass `len(x)` at the call sites, which is where this is otherwise messy. The function name may not be appropriate as I'm not intimately familiar with the rest of the code.